docs/data/charts/composition/composition.md

```diff
@@ -13,11 +13,52 @@ packageName: '@mui/x-charts'
 ## Overview
 
 The `@mui/x-charts` follows an architecture based on context providers.
-The overall idea is to pass your series and axes definitions to a single component: the `<ChartContainer />`.
-This component transforms the data and makes it available to its children.
+The overall idea is to pass your series and axes definitions to special components.
+This component transforms the data and makes it available to its children, which can be composed.
 
-Based on the data provided by the container, you can render some graphical elements with provided subcomponents, such as `<LinePlot />` or `<ChartsYAxis />`.
-Or you can [create your own components](/x/react-charts/components/).
+There are two main classes of components, which are used to create a chart.
+
+### 1. Structural components
+
+These are used to define the chart's structure and data.
+
+:::info
+Most of the demos here will be using the `ChartContainer` component.
+To see demos using the separate `ChartDataProvider` and `ChartsSurface` components, check the [HTML components documentation](/x/react-charts/components/#html-components).
+:::
+
+#### The `ChartDataProvider` and `ChartsSurface` components
+
+As the name suggests, the `ChartDataProvider` provides the data to the children components.
+While the `ChartsSurface` renders the SVG elements.
+
+```jsx
+<ChartDataProvider>
+  <ChartsSurface>{children}</ChartsSurface>
+</ChartDataProvider>
+```
```

Comment:
That's the obvious part, but the more interesting aspect is  "to which one should I pass the series, the axes, the width, height"